### ISSUE TITLE
naming_authority should be added to the related_dataset id when relation_type is parent

### ISSUE THREAD

**Issue by magnarem** [metno/py-mmd-tools#242]
The `related_dataset` element should also have value on the form `naming_autorithy:UUID` as we do for the `metadata_identifier` element.

So in `nc_to_mmd.py` in the `get_related_dataset`-function, we will need to add similar logic as on lines `874-890` in the `get_metadata_identifier`- function in the `nc_to_mmd.py`, so that lines `900-904` will be something like this:

```
for i in range(len(ids)):
    data.append({
         'id': naming_authority + ':' + ids[i],
         'relation_type': relation_types[i],
     })
```
This will make sure that the relation between parent/child is concurrent. I.e a child have `related_dataset: naming_authority:UUID` pointing to the parent with the `metadata_identifier: naming_authority:UUID`

Then DMCI will override this for dev and staging environments.

**Comment by charlienegri** [comment]
we'll need a new issue for `dmci` too, once this is done, because at the moment dmci overrides only the namespace in metadata_identifier

**Comment by charlienegri** [comment]
good point, the way the namespace replacement works in dmci for the metadata_identifier at the moment is that we append .dev or .staging to whatever the namespace is, but we could implement the replacement for the parent behind an if guard, so that it's done only if the parent namespace is no.met.. unless it's possible that a parent is on data.met.no but not in dev nor staging? maybe as you say the safest is to leave the parent untouched

**Comment by mortenwh** [comment]
In this case, I think we can only check that the related dataset id is correctly composed? I.e., check the format, and add an error to `self.missing_attributes['errors']` if it is wrong.

**Comment by magnarem** [comment]
I have an implementation in branch [fix-issue-242] (https://github.com/metno/py-mmd-tools/tree/fix-issue-242).

I tested a satellite swath dataset as it as parent/child relation with missing naming authority in the related_dataset element.
When I run `nc2mmd` the error I want is raised, but I get some additional stack-trace error, but I cannot see whats causing it.
All tests and coverages pass locally, I will create an MR to see if the pipeline fails.

Here is the command I run, and the output I get:

```
nc2mmd -u https://thredds.met.no/thredds/dodsC/remotesensingsatellite/polar-swath/2023/04/20/noaa19-avhrr-20230420212842-20230420213616.nc -o mmd/ -i /lustre/storeB/immutable/archive/projects/remotesensing/satellite-thredds/polar-swath/2023/04/20/noaa19-avhrr-20230420212842-20230420213616.nc -l log/
WARNING:root:Using default values METNCS for the MMD collection field. Please, specify other collection(s) if this is wrong. Valid collections are provided in the MMD documentation (https://htmlpreview.github.io/?https://github.com/metno/mmd/blob/master/doc/mmd-specification.html#collection-keywords)
/home/magnarem/anaconda3/envs/testenv/lib/python3.9/site-packages/py_mmd_tools/nc_to_mmd.py:1352: UserWarning: 
	CF attribute featureType is missing - one of the feature types listed in Table 9.1 in https://cfconventions.org/Data/cf-conventions/cf-conventions-1.10/cf-conventions.html#_features_and_feature_types should be used.
  warnings.warn('\n\t'+'\n\t'.join(self.missing_attributes['warnings']))
Traceback (most recent call last):
  File "/home/magnarem/anaconda3/envs/testenv/bin/nc2mmd", line 8, in <module>
    sys.exit(_main())
  File "/home/magnarem/anaconda3/envs/testenv/lib/python3.9/site-packages/py_mmd_tools/script/nc2mmd.py", line 122, in _main
    main(create_parser().parse_args())  # entry point in setup.cfg
  File "/home/magnarem/anaconda3/envs/testenv/lib/python3.9/site-packages/py_mmd_tools/script/nc2mmd.py", line 112, in main
    req_ok, msg = md.to_mmd(
  File "/home/magnarem/anaconda3/envs/testenv/lib/python3.9/site-packages/py_mmd_tools/nc_to_mmd.py", line 1354, in to_mmd
    raise AttributeError('\n\t'+'\n\t'.join(self.missing_attributes['errors']))
AttributeError: 
	related_dataset_id ACDD attribute is missing naming_authority in the identifier, relation parent.
```

**Comment by magnarem** [comment]
I was checking a bit more in this script, and have concluded that this is an expected error.

Since we have this line ` File "py_mmd_tools/nc_to_mmd.py", line 1354, in to_mmd
    raise AttributeError('\n\t'+'\n\t'.join(self.missing_attributes['errors']))`
    
It is expected that the traceback is printed when we raise an exception. The question is if the raised exceptions should be handeled somewhere else? It is why we rais an exception there in the first place.

Also, it is probably not good that the script fails if no arguments are given.

```
/py_mmd_tools/script/nc2mmd.py 
Traceback (most recent call last):
  File "/home/magnarem/git/s-enda/metno/py-mmd-tools/./py_mmd_tools/script/nc2mmd.py", line 126, in <module>
    main(create_parser().parse_args())
  File "/home/magnarem/git/s-enda/metno/py-mmd-tools/./py_mmd_tools/script/nc2mmd.py", line 84, in main
    raise ValueError('OPeNDAP url must be provided')
ValueError: OPeNDAP url must be provided
```
Here we see similar behaviour, that we raise an error, and the stacktrace are printent.
Here we probably should print the command line help?

**Comment by magnarem** [comment]
The #242 issue is solved, the id is checked and a message is printed to the user when the id is note on the correct form.

For the other issues commented in the end of #242. If we will solve this, we can create a new issue.
A solution is to modify the `nc2mmd.py` in the scripts folder, by adding try except to the calls.

```
def _main():  # pragma: no cover
    try:
        main(create_parser().parse_args())  # entry point in setup.cfg
    except ValueError as e:
        print(e)
    except AttributeError as e:
        print(e)

if __name__ == '__main__':  # pragma: no cover
    try:
        main(create_parser().parse_args())
    except ValueError as e:
        print(e)
    except AttributeError as e:
        print(e)
```

**Comment by johtoblan** [comment]
Solved by #251
